File: src/main/java/xyz/oribuin/skyblock/command/impl/SettingsCommand.java
package xyz.oribuin.skyblock.command.impl;

import dev.rosewood.rosegarden.RosePlugin;
import dev.rosewood.rosegarden.command.framework.BaseRoseCommand;
import dev.rosewood.rosegarden.command.framework.CommandContext;
import dev.rosewood.rosegarden.command.framework.CommandInfo;
import dev.rosewood.rosegarden.command.framework.annotation.RoseExecutable;
import org.bukkit.entity.Player;
import xyz.oribuin.skyblock.island.Island;
import xyz.oribuin.skyblock.manager.DataManager;

public class SettingsCommand extends BaseRoseCommand {

    public SettingsCommand(RosePlugin rosePlugin) {
        super(rosePlugin);
    }

    @RoseExecutable
    public void execute(CommandContext context) {
        DataManager manager = this.rosePlugin.getManager(DataManager.class);
        Player player = (Player) context.getSender();
        Island island = manager.getIsland(player.getUniqueId());

        if (island == null) {
            // TODO: Locale
            player.sendMessage("No Island");
            return;
        }

        player.sendMessage("imagine the settings gui opened (" + island.getSettings().toString() + ")");
    }

    @Override
    protected CommandInfo createCommandInfo() {
        return CommandInfo.builder("settings")
                .descriptionKey("command-settings-description")
                .permission("skyblock.warp.settings")
                .playerOnly(false)
                .build();
    }

}

File: Plugin/src/main/kotlin/xyz/oribuin/skyblock/command/command/SettingsCommand.kt
package xyz.oribuin.skyblock.command.command

import dev.rosewood.rosegarden.RosePlugin
import dev.rosewood.rosegarden.command.framework.CommandContext
import dev.rosewood.rosegarden.command.framework.RoseCommandWrapper
import dev.rosewood.rosegarden.command.framework.RoseSubCommand
import dev.rosewood.rosegarden.command.framework.annotation.Inject
import dev.rosewood.rosegarden.command.framework.annotation.RoseExecutable
import xyz.oribuin.skyblock.gui.SettingsGUI
import xyz.oribuin.skyblock.util.asMember
import xyz.oribuin.skyblock.util.getIsland
import xyz.oribuin.skyblock.util.getMenu
import xyz.oribuin.skyblock.util.send

class SettingsCommand(rosePlugin: RosePlugin, parent: RoseCommandWrapper) : RoseSubCommand(rosePlugin, parent) {

    @RoseExecutable
    fun execute(@Inject context: CommandContext) {
        val member = context.asMember(this.rosePlugin)
        val island = member.getIsland(this.rosePlugin)

        if (island == null) {
            member.onlinePlayer?.let { this.rosePlugin.send(it, "no-island") }
            return
        }

        this.rosePlugin.getMenu(SettingsGUI::class).openMenu(member)
    }

    override fun getDefaultName(): String = "settings"

    override fun getDescriptionKey(): String = "command-warp-settings-description"

    override fun getRequiredPermission(): String = "skyblock.command.warp.settings"

    override fun isPlayerOnly(): Boolean = true
}